ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerSSLTest.java

```diff
@@ -36,20 +41,41 @@
 public class RouterServerSSLTest {
   private static MockCluster sslCluster;
   private static RouterServerTestFramework testFramework;
+  private static MockClusterMap clusterMap;
 
   @BeforeClass
   public static void initializeTests()
       throws Exception {
     File trustStoreFile = File.createTempFile("truststore", ".jks");
+    String sslEnabledDataCentersStr = "DC1,DC2,DC3";
     Properties serverSSLProps = new Properties();
-    TestSSLUtils.addSSLProperties(serverSSLProps, "DC1,DC2,DC3", SSLFactory.Mode.SERVER, trustStoreFile, "server");
+    TestSSLUtils
+        .addSSLProperties(serverSSLProps, sslEnabledDataCentersStr, SSLFactory.Mode.SERVER, trustStoreFile, "server");
     Properties routerProps = getRouterProperties("DC1");
-    TestSSLUtils.addSSLProperties(routerProps, "DC2,DC3", SSLFactory.Mode.CLIENT, trustStoreFile, "router-client");
+    TestSSLUtils.addSSLProperties(routerProps, sslEnabledDataCentersStr, SSLFactory.Mode.CLIENT, trustStoreFile,
+        "router-client");
     MockNotificationSystem notificationSystem = new MockNotificationSystem(9);
-    sslCluster =
-        new MockCluster(notificationSystem, true, "DC1,DC2,DC3", serverSSLProps, false, SystemTime.getInstance());
+    sslCluster = new MockCluster(notificationSystem, serverSSLProps, false, SystemTime.getInstance());
     sslCluster.startServers();
-    testFramework = new RouterServerTestFramework(routerProps, sslCluster, notificationSystem);
+    clusterMap = sslCluster.getClusterMap();
+    testFramework = new RouterServerTestFramework(routerProps, clusterMap, notificationSystem);
+  }
+
+  @Before
+  public void before() {
+    for (MockDataNodeId m : clusterMap.getDataNodes()) {
+      m.setSslPortRequestedStatus(false);
+    }
+  }
+
+  @After
+  public void after() {
+    for (MockDataNodeId m : clusterMap.getDataNodes()) {
+      if (m.sslPortWasRequested()) {
```

Comment:
@nsivabalan @vgkholla's comment is that this check does not necessarily ensure that the router communicated over SSL to the datanodes - because the ssl ports could be requested by the servers as part of replication - and in these tests they all share the clustermap.